pkgs/development/python-modules/huawei-solar/default.nix

```diff
@@ -0,0 +1,55 @@
+{
+  backoff,
+  buildPythonPackage,
+  fetchFromGitLab,
+  lib,
+  hatchling,
+  hatch-vcs,
+  pytz,
+  pymodbus,
+  pyserial-asyncio,
+  typing-extensions,
+  pytest-asyncio,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "huawei-solar";
+  version = "2.3.0";
+  pyproject = true;
+
+  src = fetchFromGitLab {
+    owner = "Emilv2";
+    repo = pname;
+    rev = version;
+    hash = "sha256-PcpyyEH3Ad9oyr4aPlUgxU5S/NPoIDUZj+Ncs7FXhVA=";
+  };
+
+  build-system = [
+    hatchling
+    hatch-vcs
+  ];
+
+  dependencies = [
+    backoff
+    pytz
+    pymodbus
+    typing-extensions
+    pyserial-asyncio
+  ];
+
+  nativeCheckInputs = [
+    pytest-asyncio
+    pytestCheckHook
+  ];
+
+  meta = with lib; {
+    description = "Python library for connecting to Huawei SUN2000 Inverters over Modbus";
+    homepage = "https://gitlab.com/Emilv2/huawei-solar/";
+    changelog = "https://gitlab.com/Emilv2/huawei-solar/-/tags/${version}";
+    maintainers = with maintainers; [ Toomoch ];
+    license = licenses.agpl3Only;
+    broken = lib.versionAtLeast pymodbus.version "3.7.0";
```

Comment:
```suggestion
```
We don't have any override for this and the build will fail because of https://gitlab.com/Emilv2/huawei-solar/-/blob/22e7caac1e5483507bea92a4447d79fd7d2708ad/pyproject.toml#L28 anyway

Reply:
WDYM? The python package set has 3.7.4. The home-assistant python package set has 3.6.9.